src/api/db/models/Project.ts

```diff
@@ -0,0 +1,746 @@
+    }
+  }
+
+  static async reMapImagesToDeps({
+    projId,
+    camConfig,
+  }: {
+    projId: string;
+    camConfig: HydratedCameraConfig;
+  }) {
+    try {
+      await retry(
+        async () => {
+          // build array of operations from camConfig.deployments:
+          // for each deployment, build filter, build update, then perform bulkWrite
+          // NOTE: this function expects deps to be in chronological order!
+          const operations = [];
+          for (const [index, dep] of camConfig.deployments.entries()) {
+            const createdStart = dep.startDate || null;
+            const createdEnd = camConfig.deployments[index + 1]
+              ? camConfig.deployments[index + 1].startDate
+              : null;
+
+            const filter: Record<any, any> = { projectId: projId, cameraId: camConfig._id };
+            if (createdStart || createdEnd) {
+              filter.dateTimeOriginal = {
+                ...(createdStart && { $gte: createdStart }),
+                ...(createdEnd && { $lt: createdEnd }),
+              };
+            }
+
+            for await (const img of Image.find(filter)) {
+              const update: Partial<ImageSchema> = {};
+              if (img.deploymentId.toString() !== dep._id!.toString()) {
+                update.deploymentId = dep._id;
+              }
+
+              if (img.timezone !== dep.timezone) {
+                const dtOriginal = DateTime.fromJSDate(img.dateTimeOriginal as any as Date).setZone(
+                  img.timezone,
+                );
+                const newDT = dtOriginal.setZone(dep.timezone, { keepLocalTime: true });
+                update.dateTimeOriginal = newDT.toJSDate();
+                update.timezone = dep.timezone;
+              }
+
+              if (Object.entries(update).length > 0) {
+                const op = {
+                  updateOne: { filter: { _id: img._id }, update },
+                };
+                operations.push(op);
+              }
+            }
+          }
+          await Image.bulkWrite(operations);
+        },
+        { retries: 3 },
+      );
+    } catch (err) {
+      if (err instanceof GraphQLError) throw err;
+      throw new InternalServerError(err as string);
+    }
+  }
+
+  static async createDeploymentTask(
+    input: gql.CreateDeploymentInput,
+    context: Context,
+  ): Promise<HydratedDocument<TaskSchema>> {
+    try {
+      return await TaskModel.create(
+        {
+          type: 'CreateDeployment',
+          projectId: context.user['curr_project'],
+          user: context.user.sub,
+          config: input,
+        },
+        context,
+      );
+    } catch (err) {
+      if (err instanceof GraphQLError) throw err;
+      throw new InternalServerError(err as string);
+    }
+  }
+
+  // NOTE: this function is called by the task handler
+  static async createDeployment(
+    input: gql.CreateDeploymentInput,
+    context: Context,
+  ): Promise<HydratedCameraConfig> {
```

Comment:
Correct me if I'm wrong here, but because `CameraConfigs` are subdocuments of `Projects`, shouldn't we return `Promise<HydratedSingleSubdocument<CameraConfigSchema>>>`, in the same way we do in `CreateView()` above?

Reply:
I think that's probably more correct, however I admit that I don't have a full grasp on what exactly the _functional_ differences are between `HydratedSingleSubdocument` and `HydratedDocument`